### ISSUE TITLE
[feature] provide environmental variable for storage.path

### ISSUE THREAD

**Issue by fulara** [conan-io/conan#6831]
There seem to be a env variable for most of the things in conan.conf - why not for storage.data ?
I am forced to use sth like:
```
previous=$(conan config get storage.path)
# do_my_stuff
conan config set storage.path=${previous}
```
In one of my flows.
Env variable would be slightly more elegant from my perspective.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by krase** [comment]
Yes. That would be very helpful. I often use different `$CONAN_USER_HOME` with shared `storage.path`. Setting both in the same way would make it easier.

**Comment by memsharded** [comment]
We are actually moving in the different direction in Conan 2.0:

- Only using env-vars for the very strict minimum, like CONAN_HOME (yes different name), and some profile default
- The rest of the configuration will be file-only. 
- The idea is that configuration files are explicit, easily manageable with ``conan config install`` (while env-vars not), and they are also jinja templates. So if you want to make explicit that something must be defined in env-vars, you can do ``os.getenv(MYSTORAGE_PATH_VAR)`` in the config file.
- In any case, env-vars are no longer used internally in Conan to propagate information, that was problematic (as mostly any recipe could change those env-vars values very easily)

**Comment by memsharded** [comment]
Closing as solved in 2.0
